### PR TITLE
[otbn,dv] Fix some warnings in the OTBN DV build

### PR DESCRIPTION
**Pull request by rswarbrick**
See commit messages for details.

### CODE REVIEW

**Review by imphil — APPROVED** [review]
LGTM.

I had a look at my favorite part of the SV spec, the type inference, and here's what I found:

grafik

So I *think*, `'d1` should be sufficient to get an unsigned "1", `WIDTH'd1` isn't needed (even though it doesn't hurt).

They repeat that statement in slightly different words further down:

grafik


Furthermore: "The result [of a shift operation] signedness is determined by the left-hand operand and the remainder of the expression, as outlined in 11.8.1."

**Comment by rswarbrick** [comment]
Ah, I see. Your suggestion works because `1 << 31` *is* a valid unsigned 32-bit number. Of course, it *won't* work if we need a 33 bit number anywhere.

In my free time, I've spent literally hours staring at section 11 of IEEE-1800. It's really hard to follow! Especially all the stuff about self-determined vs. non-self-determined bit lengths.
